src/main/java/it/gov/pagopa/pu/bff/connector/debt_position/client/ReceiptClient.java

```diff
@@ -0,0 +1,54 @@
+package it.gov.pagopa.pu.bff.connector.debt_position.client;
+
+import it.gov.pagopa.pu.bff.connector.debt_position.config.DebtPositionApisHolder;
+import it.gov.pagopa.pu.bff.dto.ReceiptViewFiltersDTO;
+import it.gov.pagopa.pu.bff.util.PageUtils;
+import it.gov.pagopa.pu.debtpositions.dto.generated.PagedModelReceiptView;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+
+@Service
+@Slf4j
+public class ReceiptClient {
+
+  private final DebtPositionApisHolder debtPositionApisHolder;
+
+  public ReceiptClient(DebtPositionApisHolder debtPositionApisHolder) {
+    this.debtPositionApisHolder = debtPositionApisHolder;
+  }
+
+  public PagedModelReceiptView getReceipts(ReceiptViewFiltersDTO receiptViewFiltersDTO, Pageable pageable, String accessToken) {
+    try {
+      return debtPositionApisHolder.getReceiptViewSearchControllerApi(accessToken)
+        .crudReceiptsViewFindReceiptsByFilters(
+          String.valueOf(receiptViewFiltersDTO.getOrganizationId()),
+          receiptViewFiltersDTO.getReceiptOrigin().toString(),
+          receiptViewFiltersDTO.getOperatorExternalUserId(),
+          receiptViewFiltersDTO.getIuv(),
+          receiptViewFiltersDTO.getIur(),
+          receiptViewFiltersDTO.getIud(),
+          receiptViewFiltersDTO.getDebtPositionTypeOrgId(),
+          receiptViewFiltersDTO.getFromDate(),
+          receiptViewFiltersDTO.getToDate(),
+          PageUtils.getPageNumber(pageable),
+          PageUtils.getPageSize(pageable),
+          PageUtils.getSortList(pageable));
+    } catch (HttpClientErrorException e) {
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
+        log.warn("Receipts for organizationId {} not found", receiptViewFiltersDTO.getOrganizationId());
+        return null;
+      }
+      log.error("Error retrieving receipts for organizationId: {}", receiptViewFiltersDTO.getOrganizationId(), e);
+      throw e;
+    } catch (Exception e) {
+      log.error("Unexpected error while retrieving receipts for organizationId: {}", receiptViewFiltersDTO.getOrganizationId(), e);
+      throw e;
+    }
```

Comment:
Here you cannot expect 404! if not result, it should return an empty list, not a null body...